### PR TITLE
[libc++][NFC] Create and use test-defined simple_view concept

### PR DESCRIPTION
**Pull request by hawkinsw**
Instead of using a concept defined in the internal implementation, use a definition of the simple_view ranges concept separately defined and included in test code.

### CODE REVIEW

File: libcxx/test/std/ranges/range.adaptors/range.take/begin.pass.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ constexpr bool test() {
     ASSERT_SAME_TYPE(decltype(tv.begin()), std::counted_iterator<int*>);
   }
 
-  // __simple_view<V> && sized_range<V> && !size_range<!V>
+  // simple_view<V> && sized_range<V> && !size_range<!V>
```
Comment (cjdb) [inline libcxx/test/std/ranges/range.adaptors/range.take/begin.pass.cpp:72]: This comment should use the standard name, _`simple-view`_.

File: libcxx/test/support/test_range.h [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 #include <iterator>
 #include <ranges>
+#include <__concepts/same_as.h>
```
Comment (cjdb) [inline libcxx/test/support/test_range.h:14]:
```suggestion
#include <concepts>
```

File: libcxx/test/std/ranges/range.adaptors/range.drop/types.h [hunk 3]
```diff
@@ -15,21 +15,21 @@
 int globalBuff[8];
 
 template <class T>
-struct sentinel {
+struct drop_sentinel {
```
Comment (cjdb) [inline libcxx/test/std/ranges/range.adaptors/range.drop/types.h:18]: This change is unrelated to the commit in question. Is there a reason why it's being added here?

**Review by cjdb — CHANGES_REQUESTED** [review]
Thanks for working on this. Not much to go, and I'd like to see this approved by the end of today if possible (merged too, but that's at the whim of CI).

File: libcxx/test/support/test_range.h [hunk 4]
```diff
@@ -82,4 +83,10 @@ using NonBorrowedView = std::ranges::single_view<int>;
 static_assert(std::ranges::view<NonBorrowedView>);
 static_assert(!std::ranges::borrowed_range<NonBorrowedView>);
 
+template <class Range>
```
Comment (ldionne) [inline libcxx/test/support/test_range.h:86]: Should we keep around some kind of static assert to ensure that the test definition of `simple_view` doesn't diverge from the libc++ definition with `__simple_view`?

Comment (hawkinsw) [reply libcxx/test/support/test_range.h:86]: I think that is a *great* idea!

Comment (hawkinsw) [reply libcxx/test/support/test_range.h:86]: I will propose something in the follow up. Sorry for the delay!

File: libcxx/test/std/ranges/range.adaptors/range.drop/types.h (hunk 3 above)
Comment (hawkinsw) [reply libcxx/test/std/ranges/range.adaptors/range.drop/types.h:18]: This is needed because there is a `sentinel` type in the `test_range.h` file that causes a conflict when it is included to get the new `simple_view` concept. Do you have thoughts on how to handle it?

Comment (cjdb) [reply libcxx/test/std/ranges/range.adaptors/range.drop/types.h:18]: Yes, but it's a whole-project change, so this is fine :)

Comment (hawkinsw) [reply libcxx/test/std/ranges/range.adaptors/range.drop/types.h:18]: I completely understand!! Thanks!

File: libcxx/test/support/test_range.h (hunk 2 above)
Comment (hawkinsw) [reply libcxx/test/support/test_range.h]: Thank you!!

File: libcxx/test/support/test_range.h (hunk 4 above)
Comment (hawkinsw) [reply libcxx/test/support/test_range.h:86]: Let me know what you think.

**Comment by hawkinsw** [comment]
Sorry! Do not review yet. I screwed something up. Sorry!

File: libcxx/test/std/ranges/simple_view_consistent.pass.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
```
Comment (cjdb) [inline libcxx/test/std/ranges/simple_view_consistent.pass.cpp:1]: Since this is a meta-test, please move this file into `test/libcxx`.

**Review by cjdb — APPROVED** [review]
LGTM pending my one comment. Thanks for taking the initiative on this!

File: libcxx/test/std/ranges/simple_view_consistent.pass.cpp (hunk 5 above)
Comment (hawkinsw) [reply libcxx/test/std/ranges/simple_view_consistent.pass.cpp:1]: Will do!! I honestly had no idea where it should go!

Comment (hawkinsw) [reply libcxx/test/std/ranges/simple_view_consistent.pass.cpp]: I found `libcxx/test/libcxx/ranges/range.utility.helpers/simple_view.compile.pass.cpp`! I added the tests for `simple_view` there (using the same infrastructure defined for testing `std::ranges::__simple_view`). I removed the other test code. 

Because this was a bigger change than just moving the file, I was hoping you could take another look. 

Sorry!

**Review by cjdb — APPROVED** [review]
